management/server/setupkey.go

```diff
@@ -248,7 +248,7 @@ func (am *DefaultAccountManager) CreateSetupKey(ctx context.Context, accountID s
 
 	err = am.Store.ExecuteInTransaction(ctx, func(transaction Store) error {
 		if err = validateSetupKeyAutoGroups(ctx, transaction, accountID, autoGroups); err != nil {
-			return err
+			return status.Errorf(status.InvalidArgument, "invalid auto groups: %v", err)
```

Comment:
doesn't validateSetupKeyAutoGroups already return a `status.Errorf`? And the error is not wrapped, so the internal status code gets lost

Reply:
Thats done on purpose. The internal status is not found in the database, but at this level I want to return invalid argument so the user gets a proper http status code. Not 404